### DISCUSSION TITLE
[CodeStyle] Field attributes

### DISCUSSION THREAD

**Discussion by PhoebeMitchell** [Studio-Lovelies/GG-JointJustice-Unity#169] [Ideas]
> Use the [Tooltip(“This is shown on hover”)] tags to give designers more info for the variables

I think this is redundant in a lot of cases, and I always forget to update them. Suggest only using tooltips if necessary, instead of forcing "Drag the DialogueController here".

> Use [ReadOnly] tag to make the value not editable in the editor. Good for debugging

This is a custom tag as far as I can tell, and I think it's not currently used. Not a huge fan of it, as the Unity inspector already allows viewing fields in debug mode.

> When creating a variable that needs to be exposed to the Unity editor, use [SerializeField] in front of it on the same line

Personally I put [SerializeField] on the same line and put no blank line between fields. If we remove [Tooltip]s will we still have blank lines between fields? I think that looks real messy, but then where do other attributes go if we need a [Tooltip] or [Range]? I like one-lining attributes too. How about headers? I feel like these are better separate because they work as headers in code too, but should they have a blank line below them? Maybe the below code is a compromise? 

```
[Header("My Header")]
[SerializeField]
private int myInt;

[SerializeField, Tooltip("This is a field"), Range(0, 100]
private int myOtherInt;
```

**Comment by ViMaSter** [comment]
100% agree with the first two and solid hint about the Inspector debug mode! 
![image_10_007](https://user-images.githubusercontent.com/1689033/151001720-2b103656-9479-450b-9346-8aa22fb2d9be.png)

---

I don't think I have a preference for blank lines between fields as long as it's either none or one. Splitting them up can be useful, if they're grouped somehow (something like constants, one line free, config values next, another line free, references).

Regarding attributes, I actually am content with either of the two examples you provided; either each on it's own line or having them inside one set of `[]`s.  
My only issue is that my IDE doesn't allow specific attributes on their own line while keeping others grouped.

It's either
`[Header("Head"), Foo(), Bar()]`
OR
```
[Header("Head")]
[Foo]
[Bar] 
```

with no option for
```
[Header("Head")]
[Foo, Bar] 
```
Realistically I'm lazy and forgetful. If we decide to go for a mix, there's a high chance you'll have to remind me, but would totally accept it, if that's what we want to go for. : D

Given that, I think I'd go for each attribute on it's own line.
